### PR TITLE
Design and develop new page for display endpoints in statistics page

### PR DESCRIPTION
**Pull request by sajithab**
jira -https://wso2.org/jira/browse/APPM-633

### CODE REVIEW

File: modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag [hunk 1]
```diff
@@ -184,7 +184,31 @@ var usage = [];
 
             print(webapps);
 
-		}else if (action == "getAPIUsageByUser") { /* Overall Web Application Usage Graph */
+		}
+
+	  else if (action == "getAppsByEndPoint") {
```
Comment (dinusha-dilrukshi) [inline modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag]: Formatting issues.

File: modules/jaggery-apps/app-publisher-web/assets/default/assets.jag [hunk 2]
```diff
@@ -230,6 +231,18 @@ routeMnger.register('GET', 'publisher', '/publisher/assets/{operation}/{type}/',
...
     }
+    if (operation == 'app-by-endpoint') {
+                    try {
+                        caramel.render({op: operation,cuser: server.current(session),shortName: type,title: type});
+
+                    } catch (e) {
+                        response.sendError(404, 'Asset type not found'); //TODO: Replace
+                    }
+
+                    /*for each(var artifact in artifacts){
```
Comment (jenananthan) [inline modules/jaggery-apps/app-publisher-web/assets/default/assets.jag]: Remove unwanted comment

File: modules/jaggery-apps/app-publisher-web/assets/default/assets.jag [hunk 3]
```diff
@@ -230,6 +231,18 @@ routeMnger.register('GET', 'publisher', '/publisher/assets/{operation}/{type}/',
                  print(artifact.id+'<br/><br/>');
                  }*/
     }
+    if (operation == 'app-by-endpoint') {
+                    try {
+                        caramel.render({op: operation,cuser: server.current(session),shortName: type,title: type});
+
+                    } catch (e) {
+                        response.sendError(404, 'Asset type not found'); //TODO: Replace
```
Comment (jenananthan) [inline modules/jaggery-apps/app-publisher-web/assets/default/assets.jag]: Remove todo,if it is implemented correctly

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            + '/getAppsByEndPoint/',
+        type: 'POST',
+        data: {
+            'startDate': from,
+            'endDate': to
+        },
+
+        success: function (response) {
+
+//            drawSubscribedAPIsByUsers('[["wso1", "1.0.0", "http://localhost:8280/wso1/1.0.0"],' +
```
Comment (jenananthan) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js]: remove unused comments

File: modules/jaggery-apps/app-publisher-web/assets/default/assets.jag [hunk 5]
```diff
@@ -180,8 +180,9 @@ routeMnger.register('GET', 'publisher', '/publisher/assets/{type}/', function(co
 		sortOptions:{sortBy: sortby, sort: sortOpposite, icon: sortIcon}
 		});
 	} catch (e) {
-		response.sendError(400, "Your request is malformed");
+		response.sendError(400, "Your request is malformed...");
 		log.error(e.message);
+		log.error(e.stack);
```
Comment (jenananthan) [inline modules/jaggery-apps/app-publisher-web/assets/default/assets.jag]: log error with custom message e.g log.erro("custom error message ",e)

File: modules/jaggery-apps/app-publisher-web/themes/appm/partials/app-by-endpoint.hbs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                                </div>
+                                <span class="add-on-cal"><i style="margin-right:0px;" class="icon-calendar"></i></span>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+
+
+                <div class="clearfix"></div>
+                <!--<div class="form-group form-inline" id="searchUserForm">-->
```
Comment (jenananthan) [inline modules/jaggery-apps/app-publisher-web/themes/appm/partials/app-by-endpoint.hbs]: remove unwanted comments

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+function drawGraphs() {
+
+    var url = window.location.pathname;
+    var comps = url.split('/');
+    var type = comps[comps.length - 2];
+    var operation = comps[comps.length - 3];
+    var action = 'getSubscriberCountByAPIs';
+    var dateRange = $('#date-range').val();
+    var from = dateRange.split('to')[0].trim() + ":00";
+    var to = dateRange.split('to')[1].trim() + ":00";
```
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js]: Is 'dateRange' checked for mandatory prior to this level?
If not check for null below splitting

File: modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag (hunk 1 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag]: Fixed

File: modules/jaggery-apps/app-publisher-web/assets/default/assets.jag (hunk 5 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/assets/default/assets.jag]: Fixed

File: modules/jaggery-apps/app-publisher-web/assets/default/assets.jag (hunk 3 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/assets/default/assets.jag]: Fixed

File: modules/jaggery-apps/app-publisher-web/assets/default/assets.jag (hunk 2 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/assets/default/assets.jag]: Fixed

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js (hunk 4 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js]: Fixed

File: modules/jaggery-apps/app-publisher-web/themes/appm/partials/app-by-endpoint.hbs (hunk 6 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/themes/appm/partials/app-by-endpoint.hbs]: Fixed

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js (hunk 7 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/logic/flot.stat.app-endpoint.js]: Date range can't be null. Its picked by using dtpickers.

File: modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag [hunk 8]
```diff
@@ -30,7 +30,7 @@ require('/modules/publisher.js').exec(function (ctx) {
         }
```
Comment (mohanvive) [inline modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag]: remove unnecessary line spaces..

Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag]: There are many formatting issues in this jag file which is not relevant to this PR and also this formatting issue is not relevant to this PR. So rise a public jira to format whole jag file. - https://wso2.org/jira/browse/APPM-1251
And also this file git repo had been changed to the carbon-appmgt. https://github.com/wso2/carbon-appmgt/blob/master/features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/assets_api_router.jag

File: modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag [hunk 9]
```diff
@@ -184,7 +184,31 @@ var usage = [];
...
+						 error:true
+					 };
+				} else {
+					 obj = {
+						 error:false,
+						 webapps:webapps
+					 };
+				}
+				print(webapps);
+
```
Comment (jranabahu) [inline modules/jaggery-apps/app-publisher-web/apis/v1/assets_api_router.jag]: Please format these code correctly.

**Pull request merged**